mypy_ls/plugin.py

```diff
@@ -26,6 +27,12 @@
 
 tmpFile: Optional[IO[str]] = None
 
+# In non-live-mode the file contents aren't updated.
+# Returning an empty diagnostic clears the diagnostic result,
+# so store a cache of last diagnostics for each file a-la the pylint plugin,
+# so we can return some potentially-stale diagnostics.
+# https://github.com/palantir/python-language-server/blob/0.36.2/pyls/plugins/pylint_lint.py#L52-L59
```

Comment:
Could you link to the equivalent in [python-lsp-server](https://github.com/python-lsp/python-lsp-server)?

Reply:
I'll just drop the link here for easy copy.
https://github.com/python-lsp/python-lsp-server/blob/v1.0.1/pylsp/plugins/pylint_lint.py#L55-L62

Reply:
```suggestion
# https://github.com/python-lsp/python-lsp-server/blob/v1.0.1/pylsp/plugins/pylint_lint.py#L55-L62
```